File: frontend/src/services/moduleService.js
import { request } from './apiClient';

export const moduleService = {
  getModules: (courseId) => request(`/courses/${courseId}/modules`),
  getDeletedModules: (courseId) => request(`/courses/${courseId}/modules/deleted`),
  createModule: (courseId, data) => request(`/courses/${courseId}/modules`, {
    method: 'POST',
    body: JSON.stringify(data)
  }),
  updateModule: (id, data) => request(`/modules/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data)
  }),
  deleteModule: (id) => request(`/modules/${id}`, {
    method: 'DELETE'
  }),
  restoreModule: (id) => request(`/modules/${id}/restore`, {
    method: 'PUT'
  }),
  reorderModules: (orderedIds) => request('/modules/reorder', {
    method: 'PUT',
    body: JSON.stringify(orderedIds)
  })
};

export default moduleService;

File: backend/src/main/java/com/coursemanager/service/ModuleService.java
package com.coursemanager.service;

import com.coursemanager.dto.request.ModuleRequest;
import com.coursemanager.dto.response.ModuleResponse;

import java.util.List;

public interface ModuleService {

    List<ModuleResponse> getActiveModulesByCourse(Long courseId);

    List<ModuleResponse> getDeletedModulesByCourse(Long courseId);

    ModuleResponse getModuleById(Long id);

    ModuleResponse createModule(Long courseId, ModuleRequest request);

    ModuleResponse updateModule(Long id, ModuleRequest request);

    void softDeleteModule(Long id);

    void restoreModule(Long id);

    void reorderModules(List<Long> orderedIds);
}
